kyuubi-common/src/test/scala/org/apache/kyuubi/service/authentication/JdbcAuthenticationProviderImplSuite.scala

```diff
@@ -0,0 +1,143 @@
+      jdbcUrl
+        + ";user=" + dbUser
+        + ";password=" + dbPasswd,
+      datasourceProperties)
+
+    conn.prepareStatement("CREATE SCHEMA " + dbUser).execute();
+
+    conn.prepareStatement("CREATE TABLE user_auth (" +
+      "username VARCHAR(64) NOT NULL PRIMARY KEY, " +
+      "passwd VARCHAR(64))").execute();
+
+    val insertStmt = conn.prepareStatement("INSERT INTO user_auth " +
+      "(username, passwd) VALUES (?,?)")
+    insertStmt.setString(1, authUser)
+    insertStmt.setString(2, authPasswd)
+    insertStmt.execute();
+
+    conf = genJdbcAuthConfigs
+  }
+
+  override def afterAll(): Unit = {
+    super.afterAll()
+
+    // shutdown derby database
+    try {
+      DriverManager.getConnection(s"jdbc:derby:;databaseName=$authDb;shutdown=true")
+    } catch {
+      case e: Throwable =>
+    }
+  }
+
+  test("authenticate tests") {
+    var providerImpl = new JdbcAuthenticationProviderImpl(conf)
+
+    providerImpl.authenticate(authUser, authPasswd)
+
+    val e1 = intercept[AuthenticationException](providerImpl.authenticate("", ""))
+    assert(e1.getMessage.contains("user is null"))
+
+    val e2 = intercept[AuthenticationException](providerImpl.authenticate("kyuubi", ""))
+    assert(e2.getMessage.contains("password is null"))
+
+    val e4 = intercept[AuthenticationException](
+      providerImpl.authenticate(authPasswd, "pass"))
+    assert(e4.isInstanceOf[AuthenticationException])
+
+    conf = genJdbcAuthConfigs
+    conf.unset(AUTHENTICATION_JDBC_URL)
+    val e5 = intercept[IllegalArgumentException](new JdbcAuthenticationProviderImpl(conf))
+    assert(e5.getMessage.contains("JDBC url is not configured"))
+
+    conf = genJdbcAuthConfigs
+    conf.unset(AUTHENTICATION_JDBC_USERNAME)
+    val e6 = intercept[IllegalArgumentException](new JdbcAuthenticationProviderImpl(conf))
+    assert(e6.getMessage.contains("JDBC username or password is not configured"))
+
+    conf = genJdbcAuthConfigs
+    conf.unset(AUTHENTICATION_JDBC_PASSWORD)
+    val e7 = intercept[IllegalArgumentException](new JdbcAuthenticationProviderImpl(conf))
+    assert(e7.getMessage.contains("JDBC username or password is not configured"))
+
+    conf = genJdbcAuthConfigs
+    conf.unset(AUTHENTICATION_JDBC_QUERY)
+    val e8 = intercept[IllegalArgumentException](new JdbcAuthenticationProviderImpl(conf))
+    assert(e8.getMessage.contains("Query SQL is not configured"))
+
+    conf.set(
+      AUTHENTICATION_JDBC_QUERY,
+      "INSERT INTO user_auth (username, password) " +
+        " VALUES ('demouser','demopassword'); ")
+    val e9 = intercept[IllegalArgumentException](new JdbcAuthenticationProviderImpl(conf))
+    assert(e9.getMessage.contains("Query SQL must start with \"SELECT\""))
+
+    conf.unset(AUTHENTICATION_JDBC_URL)
+    val e10 = intercept[IllegalArgumentException](new JdbcAuthenticationProviderImpl(conf))
+    assert(e10.getMessage.contains("JDBC url is not configured"))
+  }
+
+  private def genJdbcAuthConfigs: KyuubiConf = {
```

Comment:
let's init the memeber variable `conf` and use `conf.copy()` instead of call this method every time

Reply:
agree.